internal/pkg/file/es.go

```diff
@@ -105,73 +66,75 @@ func GetFileDoc(ctx context.Context, bulker bulk.Bulk, uploadID string) ([]es.Hi
-		req.DocumentID = fmt.Sprintf("%s.%d", docID, chunkNum)
-		if req.Header == nil {
-			req.Header = make(http.Header)
-		}
-		req.Header.Set("Content-Type", "application/cbor")
-		req.Header.Set("Accept", "application/json")
-		req.Refresh = "true"
-	})
+// Retrieves a file Metadata as an Info object
+func GetInfo(ctx context.Context, bulker bulk.Bulk, indexPattern string, uploadID string) (Info, error) {
+	results, err := GetMetadata(ctx, bulker, indexPattern, uploadID)
 	if err != nil {
-		return err
+		return Info{}, err
+	}
+	if len(results) == 0 {
+		return Info{}, ErrInvalidID
+	}
+	if len(results) > 1 {
+		return Info{}, fmt.Errorf("unable to locate upload record, got %d records, expected 1", len(results))
 	}
 
-	var response ChunkUploadResponse
-	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return err
+	var fi MetaDoc
+	if err := json.Unmarshal(results[0].Source, &fi); err != nil {
+		return Info{}, fmt.Errorf("file meta doc parsing error: %w", err)
 	}
-	log.Trace().Int("statuscode", resp.StatusCode).Interface("chunk-response", response).Msg("uploaded chunk")
 
-	if response.Error.Type != "" {
-		return fmt.Errorf("%s: %s caused by %s: %s", response.Error.Type, response.Error.Reason, response.Error.Cause.Type, response.Error.Cause.Reason)
+	// calculate number of chunks required
+	cnt := fi.File.Size / fi.File.ChunkSize
+	if fi.File.Size%fi.File.ChunkSize > 0 {
+		cnt += 1
 	}
-	return nil
+
+	return Info{
+		ID:        fi.UploadID,
+		Source:    fi.Source,
+		AgentID:   fi.AgentID,
+		ActionID:  fi.ActionID,
+		DocID:     results[0].ID,
+		ChunkSize: fi.File.ChunkSize,
+		Total:     fi.File.Size,
+		Count:     int(cnt),
+		Start:     fi.Start,
+		Status:    Status(fi.File.Status),
+	}, nil
 }
 
-type ChunkUploadResponse struct {
-	Index   string `json:"_index"`
-	ID      string `json:"_id"`
-	Result  string `json:"result"`
-	Version int    `json:"_version"`
-	Shards  struct {
-		Total   int `json:"total"`
-		Success int `json:"successful"`
-		Failed  int `json:"failed"`
-	} `json:"_shards"`
-	Error es.ErrorT `json:"error"`
+// retrieves a full chunk document, Data included
+func GetChunk(ctx context.Context, bulker bulk.Bulk, indexPattern string, source string, fileID string, chunkNum int) (Chunk, error) {
+	var chunk Chunk
+	out, err := bulker.Read(ctx, fmt.Sprintf(indexPattern, source), fmt.Sprintf("%s.%d", fileID, chunkNum))
+	if err != nil {
+		return chunk, err
+	}
+	err = json.Unmarshal(out, &chunk)
+	return chunk, err
 }
 
 // Retrieves a subset of chunk document fields, specifically omitting the Data payload (bytes)
 // but adding the calculated field "size", that is the length, in bytes, of the Data field
 // the chunk's ordered index position (Pos) is also parsed from the document ID
-func GetChunkInfos(ctx context.Context, bulker bulk.Bulk, baseID string) ([]ChunkInfo, error) {
+func GetChunkInfos(ctx context.Context, bulker bulk.Bulk, indexPattern string, baseID string) ([]ChunkInfo, error) {
```

Comment:
Do the read functions here need the index pattern as a param as part of the preparation to allow more generic file reads? The upload functions (in `internal/pkg/file/uploader/es.go`) do not have an index param

Reply:
yes, the index patterns will be different for file delivery, so it became a parameter